src/attributes.jl

```diff
@@ -74,6 +74,24 @@ SetAttributeNotAllowed(attr::AnyAttribute) = SetAttributeNotAllowed(attr, "")
 operation_name(err::SetAttributeNotAllowed) = "Setting attribute $(err.attr)"
 message(err::SetAttributeNotAllowed) = err.message
 
+"""
+    struct AddAttributeNotAllowed{AttrType} <: NotAllowedError
+        attr::AttrType
+        message::String # Human-friendly explanation why the attribute cannot be set
+    end
+
+An error indicating that the attribute `attr` is supported (see
+[`supports`](@ref)) but cannot be added for some reason (see the error string).
+"""
+struct AddAttributeNotAllowed{AttrType<:AnyAttribute} <: NotAllowedError
+    attr::AttrType
+	message::String # Human-friendly explanation why the attribute cannot be set
+end
+AddAttributeNotAllowed(attr::AnyAttribute) = AddAttributeNotAllowed(attr, "")
```

Comment:
Maybe a more descriptive default message?

Reply:
There is already a more descriptive one printed by `NotAllowedError`, the message field is for custom messages.